lib/streams/compound-duplex.js

```diff
@@ -102,4 +109,37 @@ CompoundDuplex.prototype._abortStream = function() {
 	}
 };
 
+CompoundDuplex.prototype.wrap = function(stream) {
+	var self = this;
+
+
+	stream.on('error', function(error) {
+		self.emit('error', error);
+	});
+
+	if(stream.destroy) {
+		this._abortStream = function() {
+			this._internalStream.destroy();
+		};	
+	}
+
+	this._internalStream = stream;
+	this.getInternalStream = function() {
+		return this._internalStream;
+	};
+
+	if(!this._compoundWrapNoWritablePipe) {
+		// Don't pipe this compound stream from the writable side
+		this._compoundWritable.pipe(stream, { noConvert: true });
+	}
+	// Pipe Readable side through intermidiate stream, so we can act on the internal stream
+	var wrapOutputPassThrough = this._wrapOutputPassThrough = new PassThrough(this._compoundOptions);
+	wrapOutputPassThrough.pipe(this._compoundReadable);
+	stream.pipe(wrapOutputPassThrough);
```

Comment:
That is logic that should live in RequestStream, as it is specific to wanting to read out early output on error.  It should not be in CompoundDuplex and we don't need that overhead every time we wrap a streams1 stream.